gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/GetApiMetadataUseCase.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.api.use_case;
+
+import io.gravitee.apim.core.api.crud_service.ApiCrudService;
+import io.gravitee.apim.core.api.exception.ApiNotFoundException;
+import io.gravitee.apim.core.api.model.ApiMetadata;
+import io.gravitee.apim.core.api.query_service.ApiMetadataQueryService;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GetApiMetadataUseCase {
+
+    private final ApiCrudService apiCrudService;
+    private final ApiMetadataQueryService apiMetadataQueryService;
+
+    public GetApiMetadataUseCase(ApiCrudService apiCrudService, ApiMetadataQueryService apiMetadataQueryService) {
+        this.apiCrudService = apiCrudService;
+        this.apiMetadataQueryService = apiMetadataQueryService;
+    }
+
+    public Output execute(Input input) {
+        // Check that api exists with env id
+        var api = apiCrudService.get(input.apiId);
+
+        if (!api.getEnvironmentId().equals(input.envId)) {
+            throw new ApiNotFoundException(input.apiId);
+        }
+
+        Map<String, ApiMetadata> apiMetadataMap = apiMetadataQueryService.findApiMetadata(input.apiId);
+        var sortedApiMetadataList = this.sortApiMetadata(apiMetadataMap.values(), input.sortBy);
+        var filteredApiMetadataList = this.filterApiMetadata(sortedApiMetadataList, input.filterBy);
+
+        return new Output(filteredApiMetadataList);
+    }
+
+    private List<ApiMetadata> filterApiMetadata(List<ApiMetadata> apiMetadataList, String filterBy) {
+        if (filterBy != null) {
+            if (filterBy.equals("GLOBAL")) {
```

Comment:
Maybe we could introduce an internal enum in `Input` record to avoid those "hard-coded" string 🤔